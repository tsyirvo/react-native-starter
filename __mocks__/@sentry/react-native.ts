module.exports = {
  ReactNavigationInstrumentation: jest.fn().mockImplementation(() => {}),
};
