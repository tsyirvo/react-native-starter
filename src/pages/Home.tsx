import { useTranslation } from 'react-i18next';

import { Header, Informations, Version } from '$components/home';
import { Box, Button, Text } from '$components/ui/primitives';
import Screen from '$components/ui/Screen';
import { config } from '$core/constants';
import { HomeScreenNavigationProp } from '$navigation/navigation.types';

type HomeProps = {
  navigation: HomeScreenNavigationProp;
};

const Home = ({ navigation }: HomeProps) => {
  const { t } = useTranslation('homeScreen');

  const goToOtherPage = () => navigation.navigate('OtherPage');

  console.log('config', config);

  return (
    <Screen>
      <Header />

      <Box pb="global_32" px="global_24">
        <Text mt="global_32" variant="large">
          {t('navigation.title')}
        </Text>

        <Box mt="global_8">
          <Button
            alignItems="center"
            label={t('navigation.content')}
            testID="goto_otherPage"
            onPress={goToOtherPage}
          />
        </Box>

        <Informations />

        <Version />
      </Box>
    </Screen>
  );
};

export default Home;
