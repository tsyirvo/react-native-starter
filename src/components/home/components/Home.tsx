import React, { useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { ScrollView } from 'react-native';

import { HomeScreenNavigationProp } from '$routes/routes.types';
import { Flex, Text, Title, Button } from '$shared/primitives';
import SafeView from '$shared/SafeView';

import Header from './Header';
import Informations from './Informations';

type Props = {
  navigation: HomeScreenNavigationProp;
};

const Home = ({ navigation }: Props) => {
  const { t } = useTranslation();

  const goToOtherPage = useCallback(() => navigation.navigate('OtherPage'), []);

  return (
    <SafeView edges={['bottom']}>
      <ScrollView>
        <Header />

        <Flex px="large" pb={150}>
          <Title fontWeight={600} mt="large">
            {t('home.navigation_title')}
          </Title>
          <Button
            testID="goto_otherPage"
            onPress={goToOtherPage}
            alignItems="center"
            mt="medium"
          >
            <Text>{t('home.navigation_content')}</Text>
          </Button>

          <Informations />
        </Flex>
      </ScrollView>
    </SafeView>
  );
};

export default Home;
