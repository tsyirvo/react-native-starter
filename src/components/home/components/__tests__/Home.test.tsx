/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/no-explicit-any */

import { fireEvent } from '@testing-library/react-native';
import React from 'react';

import render from '$tests/utils';

import Home from '../Home';

describe('Home page component', () => {
  // given
  const props = { navigation: { navigate: jest.fn() } as any };

  it('should render correctly', () => {
    // When
    const wrapper = render(<Home {...props} />);

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger the navigate method when the button is pressed', () => {
    // When
    const { getByTestId } = render(<Home {...props} />);

    fireEvent.press(getByTestId('goto_otherPage'));

    // Then
    expect(props.navigation.navigate).toHaveBeenCalled();
  });
});
