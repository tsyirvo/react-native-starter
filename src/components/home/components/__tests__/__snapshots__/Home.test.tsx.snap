// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "height": 350,
            "width": "100%",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "header",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 350,
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <View
          alignItems="center"
          justifyContent="flex-end"
          mb={25}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "flex-end",
                "marginBottom": 25,
              },
            ]
          }
        >
          <Text
            color="white"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 24,
                  "lineHeight": 33,
                },
              ]
            }
            testID="home_title"
            variant="large"
          >
            React Native Starter
          </Text>
        </View>
      </View>
      <View
        pb={150}
        px="large"
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingBottom": 150,
              "paddingLeft": 24,
              "paddingRight": 24,
            },
          ]
        }
      >
        <Text
          color="black"
          fontWeight={600}
          mt="large"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 24,
                "fontWeight": "600",
                "lineHeight": 33,
                "marginTop": 24,
              },
            ]
          }
          variant="large"
        >
          Navigate to another page
        </Text>
        <View
          accessible={true}
          alignItems="center"
          focusable={true}
          mt="medium"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#e5e5e5",
                "borderRadius": 16,
                "marginTop": 16,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              },
            ]
          }
          testID="goto_otherPage"
          variant="base"
        >
          <Text
            color="black"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
            variant="regular"
          >
            Press to navigate
          </Text>
        </View>
        <Text
          color="black"
          fontWeight={600}
          mb={10}
          mt={25}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 24,
                "fontWeight": "600",
                "lineHeight": 33,
                "marginBottom": 10,
                "marginTop": 25,
              },
            ]
          }
          variant="large"
        >
          Storybook
        </Text>
        <Text
          color="black"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          Access the Storybook form the Dev Menu with
           Cmd+R
        </Text>
        <Text
          color="black"
          fontWeight={600}
          mb={10}
          mt={25}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 24,
                "fontWeight": "600",
                "lineHeight": 33,
                "marginBottom": 10,
                "marginTop": 25,
              },
            ]
          }
          variant="large"
        >
          Tests
        </Text>
        <Text
          color="black"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          You can run tests with Jest or Detox
        </Text>
        <Text
          color="black"
          fontWeight={600}
          mb={10}
          mt={25}
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 24,
                "fontWeight": "600",
                "lineHeight": 33,
                "marginBottom": 10,
                "marginTop": 25,
              },
            ]
          }
          variant="large"
        >
          Formatting & type checking
        </Text>
        <Text
          color="black"
          style={
            Array [
              Object {
                "color": "#000",
                "fontSize": 16,
                "lineHeight": 24,
              },
            ]
          }
          variant="regular"
        >
          The code is checked with ESLint, Prettier and TypeScript
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
